bentoml/_internal/bento/build_config.py

```diff
@@ -54,98 +62,203 @@ def _convert_python_version(py_version: t.Optional[str]) -> t.Optional[str]:
     return f"{major}.{minor}"
 
 
+def _convert_cuda_version(cuda_version: t.Optional[str]) -> t.Optional[str]:
+    if cuda_version is None or cuda_version == "":
+        return None
+
+    if isinstance(cuda_version, int):
+        cuda_version = str(cuda_version)
+
+    if cuda_version == "default":
+        cuda_version = DEFAULT_CUDA_VERSION
+
+    match = re.match(r"^(\d+)$", cuda_version)
+    if match is not None:
+        cuda_version = SUPPORTED_CUDA_VERSION[cuda_version]
+
+    return cuda_version
+
+
+def _convert_user_envars(
+    envars: t.Optional[t.Union[t.List[str], t.Dict[str, t.Any]]]
+) -> t.Optional[t.Dict[str, t.Any]]:
+    if envars is None:
+        return None
+
+    if isinstance(envars, list):
+        return {k: v for e in envars for k, v in e.split("=")}
+    else:
+        return envars
+
+
+def _envars_validator(
+    _: t.Any,
+    attribute: "Attribute[t.Union[str, t.List[str], t.Dict[str, t.Any]]]",
+    value: t.Union[str, t.List[str], t.Dict[str, t.Any]],
+) -> None:
+    if isinstance(value, str):
+        if not os.path.exists(value):
+            raise InvalidArgument("envars should be a file path to .env file")
```

Comment:
let's improve this error message? user may mistakenly put

```
env: FOO=BAR
```

Reply:
will do